pandas/core/dtypes/cast.py

```diff
@@ -564,6 +564,9 @@ def invalidate_string_dtypes(dtype_set):
 
 def coerce_indexer_dtype(indexer, categories):
     """ coerce the indexer input array to the smallest dtype possible """
+    indexer = np.array(indexer)
```

Comment:
use np.asarray; though see my other comment; this is an internal method; these are guaranteed to *already* be an integer index, if not then there is a problem in the caller.

Reply:
Thanks for the feedback. I've used `asarray` as you suggest. When you say they are guaranteed to be an integer index already, do you mean that this should be guaranteed by ensuring this contract is upheld every us case within the codebase? I can take it out if you prefer. The case where this contract was broken in `remove_unused_levels` has been resolved.